src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7848,14 +7848,14 @@ private StatementSyntax TryParseStatementStartingWithIdentifier(SyntaxList<Attri
             if (this.CurrentToken.ContextualKind == SyntaxKind.AwaitKeyword &&
                 this.PeekToken(1).Kind == SyntaxKind.ForEachKeyword)
             {
-                return this.ParseForEachStatement(attributes, ParseAwaitKeyword(MessageID.IDS_FeatureAsyncStreams));
+                return this.ParseForEachStatement(attributes, ParseAwaitKeyword(MessageID.None));
```

Comment:
:memo: I had to move those errors to binding layer, as they were affecting binding logic (which uses `BoundNode.HasAnyErrors`), but we want to bind the same way regardless of LangVer.